examples/isolated-functions/isolated_functions.bal

```diff
@@ -0,0 +1,47 @@
+import ballerina/io;
+
+const BALLERINA = "Ballerina";
+
+final string[] & readonly prefixes = ["hello", "greetings", "hello world"];
+
+// An `isolated` function, which adds a specific number of
+// greeting strings to an array.
+// The `greetings` parameter is the array to which greetings have to
+// be added. The `count` parameter indicates the number of greetings
+// to add.
+isolated function addGreetings(string[] greetings, int count) {
+    // Call another `isolated` function to build a specific number
+    // of greetings.
+    string[] builtGreetings = getGreetings(count);
+
+    // Add the greetings returned by the `isolated` function to
+    // the mutable array passed as an argument.
+    greetings.push(...builtGreetings);
+}
+
+// An `isolated` function, which accesses global immutable state and
+// returns a mutable value.
```

Comment:
You mentioned that if locks are used, we can access global mutable state right? If so, shouldn't that be mentioned?

Reply:
Yeah, but that is for [isolated module level variables](https://github.com/ballerina-platform/ballerina-lang/issues/25898) which we don't support yet.